planner/core/rule_build_key_info.go

```diff
@@ -209,18 +209,18 @@ func (p *LogicalJoin) BuildKeyInfo(selfSchema *expression.Schema, childSchema []
 }
 
 // checkIndexCanBeKey checks whether an Index can be a Key in schema.
-func checkIndexCanBeKey(idx *model.IndexInfo, columns []*model.ColumnInfo, schema *expression.Schema) expression.KeyInfo {
+func checkIndexCanBeKey(idx *model.IndexInfo, columns []*model.ColumnInfo, schema *expression.Schema) (uniqueKey, newKey expression.KeyInfo) {
 	if !idx.Unique {
-		return nil
+		return nil, nil
 	}
-	newKey := make([]*expression.Column, 0, len(idx.Columns))
 	ok := true
 	for _, idxCol := range idx.Columns {
 		// The columns of this index should all occur in column schema.
 		// Since null value could be duplicate in unique key. So we check NotNull flag of every column.
 		find := false
 		for i, col := range columns {
 			if idxCol.Name.L == col.Name.L {
+				uniqueKey = append(uniqueKey, schema.Columns[i])
```

Comment:
The problem is still not fixed? line 225 would break the loop and `find` is false, so `uniqueKey` may only contain the first column as well.

Reply:
done